otter/test/log/test_cloudfeeds.py

```diff
@@ -211,39 +223,72 @@ def _get_request(self, _type, _id, tenant_id):
         self.req['entry']['content']['event']['tenantId'] = tenant_id
         return self.req
 
-    def test_add_event(self):
+    def _perform_add_event(self, response_sequence):
         """
-        add_event pushes event by calling cloud feed with retries
+        Given a sequence of functions that take an intent and returns a
+        response (or raises an exception), perform :func:`add_event` and
+        return the result.
         """
         log = object()
         eff = add_event(self.event, 'tid', 'ord', log)
-
-        # effect is to generate UUID
-        self.assertIs(eff.intent.func, uuid.uuid4)
-        eff = resolve_effect(eff, uuid.UUID(int=0))
         uid = '00000000-0000-0000-0000-000000000000'
 
-        # effect scoped on on tenant id
-        self.assertIs(type(eff.intent), TenantScope)
-        self.assertEqual(eff.intent.tenant_id, 'tid')
-
-        # Wrapped effect is retry
-        eff = eff.intent.effect
-        self.assertEqual(
-            eff.intent.should_retry,
-            ShouldDelayAndRetry(can_retry=retry_times(5),
-                                next_interval=exponential_backoff_interval(2)))
-
-        # effect wrapped in retry is ServiceRequest
-        eff = eff.intent.effect
-        self.assertEqual(
-            eff,
-            service_request(
-                ServiceType.CLOUD_FEEDS, 'POST', 'autoscale/events',
-                headers={
-                    'content-type': ['application/vnd.rackspace.atom+json']},
-                data=self._get_request('INFO', uid, 'tid'), log=log,
-                success_pred=has_code(201)))
+        svrq = service_request(
+            ServiceType.CLOUD_FEEDS, 'POST', 'autoscale/events',
+            headers={
+                'content-type': ['application/vnd.rackspace.atom+json']},
+            data=self._get_request('INFO', uid, 'tid'), log=log,
+            success_pred=has_code(201),
+            json_response=False)
+
+        seq = [
+            (Func(uuid.uuid4), lambda _: uid),
+            (TenantScope(mock.ANY, 'tid'), nested_sequence([
+                retry_sequence(
+                    Retry(effect=svrq, should_retry=ShouldDelayAndRetry(
+                        can_retry=mock.ANY,
+                        next_interval=exponential_backoff_interval(2))),
+                    response_sequence
+                )
+            ]))
+        ]
+
+        return perform_sequence(seq, eff)
+
+    def test_add_event_succeeds_if_request_succeeds(self):
+        """
+        Adding an event succeeds without retrying if the service request
+        succeeds.
+        """
+        body = "<some xml>"
+        resp = stub_pure_response(body, 202)
```

Comment:
Sorry for chiming in late, but this test informs that 202 succeeds whereas it should be retrying. Could you change it to 201? Also, am not sure how could you test that 202 should retry since any stub response with any code will succeed.